packages/calypso-build/webpack/sass.js

```diff
@@ -57,18 +63,20 @@ module.exports.loader = ( { preserveCssCustomProperties, includePaths, prelude }
  *
  * @return {Object[]}                  styling relevant webpack plugin objects
  */
-module.exports.plugins = ( { chunkFilename, filename, minify } ) => [
-	new MiniCssExtractPluginWithRTL( {
-		chunkFilename,
-		filename,
-		rtlEnabled: true,
-	} ),
-	new FilterWarningsPlugin( {
-		// suppress conflicting order warnings from mini-css-extract-plugin.
-		// see https://github.com/webpack-contrib/mini-css-extract-plugin/issues/250
-		exclude: /mini-css-extract-plugin[^]*Conflicting order between:/,
-	} ),
-	new WebpackRTLPlugin( {
-		minify,
-	} ),
-];
+module.exports.plugins = ( { chunkFilename, filename, minify, useStyleLoader } ) =>
+	[
+		! useStyleLoader &&
+			new MiniCssExtractPluginWithRTL( {
+				chunkFilename,
+				filename,
+				rtlEnabled: true,
+			} ),
+		new FilterWarningsPlugin( {
```

Comment:
Since this is to suppress warnings coming from `MiniCssExtractPluginWithRTL`, maybe we should also prefix it with 
```suggestion
		! useStyleLoader && new FilterWarningsPlugin( {
```

?

Reply:
Unless those warnings are maybe even gone after #35866? :thinking:

Reply:
Yes, that's true, the `FilterWarningsPlugin` should be guarded by `useStyleLoader`, too. And also the `WebpackRtlPlugin`. The whole plugin setup is completely different in dev and prod.